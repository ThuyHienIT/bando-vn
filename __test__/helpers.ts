import { randomUUID } from 'crypto';

export function generateFacility(
  type: FacilityTypeEnum,
  name: string = 'Venus Meeting Room'
): FacilityItem {
  return {
    phoneNumber: '24709620',
    id: randomUUID(),
    address: '1 JALAN KILANG,S159402',
    description:
      'Current majority suffer hair born.\nReport daughter government item keep political cut. Customer apply couple. Data happen hear must almost.',
    name,
    amenities:
      'Friend ok nation international line another within.\nUp rate when sometimes probably measure with. Product town sort. Include even alone certain.',
    type: type,
    operationHours: ['8:00', '19:00'],
    offDays: ['2022-04-23T23:18:53Z'],
    thumbnail: 'https://picsum.photos/400/200',
  };
}

export function generateBooking(facilityId: string): BookingItem {
  return {
    facilityId: facilityId,
    from: '2023-04-23T10:30:00Z',
    to: '2023-04-23T11:00:00Z',
    userEmail: 'kqthang1505@gmail.com',
    id: randomUUID(),
    createdAt: '2023-02-18T22:25:56+08:00',
    updatedAt: '2023-02-18T22:25:56+08:00',
  };
}
