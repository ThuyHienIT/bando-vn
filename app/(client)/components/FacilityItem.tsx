'use client';

import { Button, Card, Typography } from 'antd';
import Image from 'next/image';
import Link from 'next/link';
import styled from 'styled-components';

import { EyeOutlined } from '@ant-design/icons';

const CardStyle = styled(Card)`
  height: 100%;
  display: flex;
  flex-direction: column;

  && .ant-card-actions {
    margin-top: auto;
  }
`;

const ImageStyle = styled(Image)`
  width: auto;
  height: auto;
`;

interface Props {
  data: FacilityItem;
}

export function FacilityItemCmp(props: Props) {
  return (
    <CardStyle
      size="small"
      cover={
        props.data.thumbnail ? (
          <ImageStyle
            alt={props.data.name}
            src={props.data.thumbnail}
            width={240}
            height={100}
          />
        ) : null
      }
      actions={[
        <Link href={`/facility/${props.data.id}`} key="details">
          <Button type="text" size="small" icon={<EyeOutlined />}>
            Details
          </Button>
        </Link>,
      ]}
    >
      <Typography.Title level={4}>{props.data.name}</Typography.Title>
      <Typography.Text ellipsis>{props.data.description}</Typography.Text>
    </CardStyle>
  );
}
