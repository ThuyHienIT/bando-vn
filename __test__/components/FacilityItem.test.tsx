import { render, screen } from '@testing-library/react';
// import Home from '../pages/index'
import '@testing-library/jest-dom';
import { FacilityItemCmp } from '@components/FacilityItem';
import { generateFacility } from '__test__/helpers';
import { FacilityTypeEnum } from '@enums';

describe('Facility Item', () => {
  it('renders facility card as expected', () => {
    const facility = generateFacility(FacilityTypeEnum.Room);
    render(<FacilityItemCmp data={facility} />);

    const bookBtn = screen.getByRole('button', { name: /book/i });
    const detailsBtn = screen.getByRole('button', { name: /details/i });
    const name = screen.getByRole('heading', { name: facility.name });

    expect(bookBtn).toBeInTheDocument();
    expect(detailsBtn).toBeInTheDocument();
    expect(name).toBeInTheDocument();
  });

  it('renders facility card unchanged', () => {
    const facility = generateFacility(FacilityTypeEnum.Room);
    facility.id = 'test-id';

    const { container } = render(<FacilityItemCmp data={facility} />);
    expect(container).toMatchSnapshot();
  });
});
