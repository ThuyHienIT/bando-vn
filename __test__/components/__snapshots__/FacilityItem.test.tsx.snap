// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Facility Item renders facility card unchanged 1`] = `
.c0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0.c0 .ant-card-actions {
  margin-top: auto;
}

.c1 {
  width: auto;
  height: auto;
}

<div>
  <div
    class="ant-card ant-card-bordered ant-card-small c0 css-dev-only-do-not-override-diro6f"
  >
    <div
      class="ant-card-cover"
    >
      <img
        alt="Venus Meeting Room"
        class="c1"
        data-nimg="1"
        decoding="async"
        height="100"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fpicsum.photos%2F400%2F200&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fpicsum.photos%2F400%2F200&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fpicsum.photos%2F400%2F200&w=640&q=75 2x"
        style="color: transparent;"
        width="240"
      />
    </div>
    <div
      class="ant-card-body"
    >
      <h4
        class="ant-typography css-dev-only-do-not-override-diro6f"
      >
        Venus Meeting Room
      </h4>
      <span
        class="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line css-dev-only-do-not-override-diro6f"
      >
        Current majority suffer hair born.
Report daughter government item keep political cut. Customer apply couple. Data happen hear must almost.
      </span>
    </div>
    <ul
      class="ant-card-actions"
    >
      <li
        style="width: 100%;"
      >
        <span>
          <a
            href="/facility/test-id"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-diro6f ant-btn-text ant-btn-sm"
              type="button"
            >
              <span
                aria-label="eye"
                class="anticon anticon-eye"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="eye"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                  />
                </svg>
              </span>
              <span>
                Details
              </span>
            </button>
          </a>
        </span>
      </li>
    </ul>
  </div>
</div>
`;
